dev/generate_protos.py

```diff
@@ -0,0 +1,203 @@
+    assert lang in ["python", "java"]
+
+    subprocess.check_call(
+        [
+            protoc_bin,
+            f"-I={protoc_include_path}",
+            f"-I={proto_dir}",
+            f"--{lang}_out={out_dir}",
+            *[f"{proto_dir}/{proto_file}" for proto_file in proto_files],
+        ]
+    )
+
+
+def apply_python_gencode_replacement(file_path):
+    content = file_path.read_text()
+
+    for old, new in python_gencode_replacements:
+        content = content.replace(old, new)
+
+    file_path.write_text(content, encoding="UTF-8")
+
+
+def _get_python_output_filename(proto_file_name):
+    return proto_file_name.split(".")[0] + "_pb2.py"
+
+
+basic_proto_files = [
+    "databricks.proto",
+    "service.proto",
+    "model_registry.proto",
+    "databricks_artifacts.proto",
+    "mlflow_artifacts.proto",
+    "internal.proto",
+    "scalapb/scalapb.proto",
+]
+uc_proto_files = [
+    "databricks_managed_catalog_messages.proto",
+    "databricks_managed_catalog_service.proto",
+    "databricks_uc_registry_messages.proto",
+    "databricks_uc_registry_service.proto",
+    "databricks_filesystem_service.proto",
+]
+facet_proto_files = ["facet_feature_statistics.proto"]
+python_proto_files = basic_proto_files + uc_proto_files + facet_proto_files
+test_proto_files = ["test_message.proto"]
+
+
+python_gencode_replacements = [
+    (
+        "from scalapb import scalapb_pb2 as scalapb_dot_scalapb__pb2",
+        "from .scalapb import scalapb_pb2 as scalapb_dot_scalapb__pb2",
+    ),
+    ("import databricks_pb2 as databricks__pb2", "from . import databricks_pb2 as databricks__pb2"),
+    (
+        "import databricks_uc_registry_messages_pb2 as databricks__uc__registry__messages__pb2",
+        "from . import databricks_uc_registry_messages_pb2 as databricks_uc_registry_messages_pb2",
+    ),
+    (
+        "import databricks_managed_catalog_messages_pb2 as databricks__managed__catalog__"
+        "messages__pb2",
+        "from . import databricks_managed_catalog_messages_pb2 as databricks_managed_"
+        "catalog_messages_pb2",
+    ),
+]
+
+
+def gen_python_protos(protoc_bin, protoc_include_path, out_dir):
+    gen_protos(
+        mlflow_protos_dir, python_proto_files, "python", protoc_bin, protoc_include_path, out_dir
+    )
+
+    gen_protos(
+        test_protos_dir, test_proto_files, "python", protoc_bin, protoc_include_path, out_dir
+    )
+
+    for file_name in python_proto_files:
+        apply_python_gencode_replacement(Path(out_dir, _get_python_output_filename(file_name)))
+
+
+def download_and_extract_protoc(version):
+    """
+    Download and extract specific version protoc tool, return extracted protoc executable file path
+    and include path.
+    """
+    assert platform.system() in [
+        "Darwin",
+        "Linux",
+    ], "The script only supports MacOS or Linux system."
+    assert platform.machine() in [
+        "x86_64",
+        "aarch64",
```

Comment:
Do we need to support `arm64` CPU as well?

Reply:
I don't have machine to test that. We can add it when we receive the use-case.